source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/detail/PropertyContactList.tsx

```diff
@@ -0,0 +1,177 @@
+      align: 'left',
+      minWidth: 60,
+      maxWidth: 60,
+      Cell: (
+        cellProps: CellProps<Api_PropertyContact, Api_Organization | Api_Person | undefined>,
+      ) => {
+        return (
+          <StyledLink
+            target="_blank"
+            rel="noopener noreferrer"
+            to={`/contact/${
+              cellProps.row.original.person
+                ? `P${cellProps.row.original.person.id}`
+                : `O${cellProps.row.original.organization?.id}`
+            }`}
+          >
+            {cellProps.row.original?.person
+              ? formatApiPersonNames(cellProps.row.original?.person)
+              : cellProps.row.original?.organization?.name ?? ''}
+          </StyledLink>
+        );
+      },
+    },
+    {
+      Header: 'Primary Contact',
+      align: 'left',
+      minWidth: 60,
+      maxWidth: 60,
+      Cell: (
+        cellProps: CellProps<Api_PropertyContact, Api_Organization | Api_Person | undefined>,
+      ) => {
+        const primaryContact = cellProps.row.original.primaryContact;
+        if (primaryContact !== null) {
+          return (
+            <StyledLink
+              target="_blank"
+              rel="noopener noreferrer"
+              to={`/contact/${`P${primaryContact.id}`}`}
+            >
+              {formatApiPersonNames(primaryContact)}
+            </StyledLink>
+          );
+        } else {
+          return <></>;
+        }
+      },
+    },
+    {
+      Header: 'Purpose',
+      align: 'left',
+      minWidth: 60,
+      maxWidth: 60,
+      Cell: (cellProps: CellProps<Api_PropertyContact, Api_Property | undefined>) => {
+        return stringToFragment(cellProps.row.original.purpose || '');
+      },
+    },
+    {
+      Header: 'Actions',
+      align: 'left',
+      sortable: false,
+      width: 25,
+      maxWidth: 25,
+      Cell: (cellProps: CellProps<Api_PropertyContact>) => {
+        const { hasClaim } = useKeycloakWrapper();
+        return (
+          <Row className="no-gutters">
+            {hasClaim(Claims.PROPERTY_EDIT) && (
+              <Col>
+                <LinkButton
+                  icon={<FaEdit size={'2rem'} />}
+                  onClick={() => cellProps.row.original.id && onEdit(cellProps.row.original.id)}
+                ></LinkButton>
+              </Col>
+            )}
+            {hasClaim(Claims.PROPERTY_DELETE) && (
+              <StyledRemoveIconButton
+                id={`contact-delete-${cellProps.row.id}`}
+                data-testid={`contact-delete-${cellProps.row.id}`}
+                onClick={() => cellProps.row.original.id && onDelete(cellProps.row.original.id)}
+                title="Delete contact"
+              >
+                <FaTrash size="2rem" />
+              </StyledRemoveIconButton>
+            )}
+          </Row>
+        );
+      },
+    },
+  ];
+
+  return columns;
+}
+
+const FilePropertiesTable: React.FunctionComponent<IPropertyContactListProps> = ({
```

Comment:
nit: I'd prefer to name the component the same as the filename - PropertyContactList